service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/CompoundEventValidator.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.fpl.service.validators;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.fpl.enums.Event;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+
+public abstract class CompoundEventValidator implements Validator {
+
+    @Autowired
+    private EventChecker eventChecker;
+
+    public List<String> validate(CaseData caseData, List<Event> events) {
+        return events.stream()
+            .flatMap(event -> {
+                List<String> groupErrors = new ArrayList<>();
+                List<String> errors = eventChecker.validate(event, caseData);
+
+                if (isNotEmpty(errors)) {
+                    groupErrors.add(String.format("In the %s section:", event.getName().toLowerCase()));
```

Comment:
Is it correct to name event name with the section? Does it work with changed event names?

Reply:
It is as currently on prod, event names change story is separate one, and should answer question about section names, otherwise we can have misleading/inconsistent naming